webpack/assets/javascripts/react_app/components/AutoComplete/AutoCompleteActions.js

```diff
@@ -3,127 +3,159 @@ import debounce from 'lodash/debounce';
 import API from '../../API';
 import { STATUS } from '../../constants';
 import { clearSpaces } from '../../common/helpers';
+import { translate as __ } from '../../common/I18n';
 import {
   AUTO_COMPLETE_REQUEST,
   AUTO_COMPLETE_SUCCESS,
   AUTO_COMPLETE_FAILURE,
   AUTO_COMPLETE_RESET,
+  AUTO_COMPLETE_DISABLED_CHANGE,
   TRIGGERS,
 } from './AutoCompleteConstants';
 
 export const getResults = ({
-  url, searchQuery, controller, trigger,
+  url, searchQuery, controller, trigger, id,
 }) => (dispatch) => {
   startRequest({
     controller,
     searchQuery,
     trigger,
     dispatch,
+    url,
+    id,
   });
-
+  if (!url) {
+    return requestFailure({
+      error: new Error(__('No API path was provided.')),
+      id,
+      dispatch,
+    });
+  }
   const path = getAPIPath({ trigger, searchQuery, url });
   return createAPIRequest({
     controller,
     path,
     searchQuery,
     trigger,
+    url,
     dispatch,
+    id,
   });
 };
 
 let createAPIRequest = ({
-  controller,
-  path,
-  searchQuery,
-  trigger,
-  dispatch,
-}) => API.get(path)
-  .then(({
-    data,
-  }) => requestSuccess({
-    data,
-    controller,
-    dispatch,
-    searchQuery,
-    trigger,
-  }))
-  .catch(error => requestFailure({ error, dispatch }));
+  controller, path, searchQuery, trigger, url, dispatch, id,
+}) =>
+  API.get(path)
+    .then(({ data }) =>
+      requestSuccess({
+        data,
+        controller,
+        dispatch,
+        searchQuery,
+        trigger,
+        url,
+        id,
+      }))
+    .catch(error => requestFailure({ error, id, dispatch }));
 
 createAPIRequest = debounce(createAPIRequest, 250);
 
 const startRequest = ({
-  controller, searchQuery, trigger, dispatch,
+  controller, searchQuery, trigger, dispatch, url, id,
 }) => {
   dispatch({
-    type: AUTO_COMPLETE_REQUEST,
+    type: `${AUTO_COMPLETE_REQUEST}/${id}`,
```

Comment:
IMHO using `${AUTO_COMPLETE_REQUEST}/${id}` as action type is overkill.
I think it is enough to dispatch:
```js
dispatch({
  type: AUTO_COMPLETE_REQUEST,
  payload: { id, ... },
})
```

Then the reducers can listen to `AUTO_COMPLETE_REQUEST` and use the `id` from the `payload`.